code/core/src/core-server/utils/StoryIndexGenerator.ts

```diff
@@ -368,9 +373,24 @@ export class StoryIndexGenerator {
     const entries: ((StoryIndexEntryWithExtra | DocsCacheEntry) & { tags: Tag[] })[] =
       indexInputs.map((input) => {
         const name = input.name ?? storyNameFromExport(input.exportName);
-        const componentPath =
-          input.rawComponentPath &&
-          this.resolveComponentPath(input.rawComponentPath, absolutePath, matchPath);
+        let componentPath;
+        if (input.rawComponentPath?.startsWith('#')) {
+          const resolvedPaths = input.rawComponentPath
+            ? resolveImports(packageJson.imports, input.rawComponentPath, [
+                'storybook',
+                'node',
+                'import',
```

Comment:
Perhaps this should also have `stories, test` like other places have? e.g. webpack config, vite config, storybook vitest plugin